metricbeat/autodiscover/builder/metrics/metrics.go

```diff
@@ -113,17 +114,18 @@ func (m *metricAnnotations) getModule(hints common.MapStr) string {
 
 func (m *metricAnnotations) getMetricSets(hints common.MapStr, module string) []string {
 	var msets []string
+	var err error
 	msets = builder.GetHintAsList(hints, m.Key, metricsets)
 
 	if len(msets) == 0 {
-		// Special handling for prometheus as most use cases rely on exporters/instrumentation.
-		// Prometheus stats can be explicitly configured if need be.
-		if module == "prometheus" {
-			msets = []string{"collector"}
-		} else {
-			msets = mb.Registry.MetricSets(module)
+		// If no metricset list is given, take module defaults
+		// fallback to all metricsets if module has no defaults
+		msets, err = m.Registry.DefaultMetricSets(module)
+		if err != nil || len(msets) == 0 {
+			msets = m.Registry.MetricSets(module)
```

Comment:
I worry that with that we enable experimental metricsets which is probably not our intention.

Reply:
Could we exclude experimental metricsets from defaults? Is there an alternative way to explicitly enable metricsets from hints?

Reply:
Do we agree to set default metricsets for all modules? If that's the case this is not a issue

Reply:
yes, we do. but if we merge this we should make sure then that we follow up with the changes until the next release.

Reply:
Well, having that this feature is experimental... we could probably live with experimental metricsets being spawned here innocent